### PR TITLE
[dhcp6relay] Fix option parsing and add dhcpv6 client messages

### PR DESCRIPTION
**Pull request by kellyyeh**
#### Why I did it
Fixed option parsing function and added missing dhcpv6 messages.
Confirm message type was not added in PR https://github.com/Azure/sonic-buildimage/pull/10486 when client side message types are checked.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [x] 202111

### CODE REVIEW

File: src/dhcp6relay/src/relay.cpp [hunk 1]
```diff
@@ -638,15 +643,17 @@ void callback(evutil_socket_t fd, short event, void *arg) {
         }
         case DHCPv6_MESSAGE_TYPE_SOLICIT:
         case DHCPv6_MESSAGE_TYPE_REQUEST: 
+	    case DHCPv6_MESSAGE_TYPE_CONFIRM:
```
Comment (yxieca) [inline src/dhcp6relay/src/relay.cpp:646]: indentation issue?

**Review by yxieca — APPROVED** [review]

File: src/dhcp6relay/src/relay.cpp (hunk 1 above)
Comment (kellyyeh) [reply src/dhcp6relay/src/relay.cpp]: Fixed

**Comment by shlomibitton** [comment]
@kellyyeh @yxieca can you please cherry-pick this to 202012 as well?

**Review by yxieca — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Pull request merged**